### ISSUE TITLE
sparsegrid spinup discontinuities

### ISSUE THREAD

**Issue by djk2120** [ESCOMP/CTSM#2859]
### Brief summary of bug

We are seeing some spinup discontinuities when running ctsm5.3.0 in a sparsegrid configuration

### Details of bug

@linniahawkins ran a small parameter perturbation ensemble and all of the simulations show jump discontinuities when transitioning from AD to SASU, and from SASU to postSASU, but not from postSASU to transient historical. We're pretty convinced that the model is reading from the correct finidat in each case, but it seems like maybe the interpolation is not operating correctly (?).

### Important output or errors that show the problem

Here's an example of a time-series of GPP that illustrates the discontinuities:
<img src="https://github.com/user-attachments/assets/fc83d555-1d2b-4ad4-b240-caeb528f23e3" alt="gpp_timeseries" width="350"/>

Below, the left panel shows that the pSASU TOTVEGC is very obviously connnected to the SASU TOTVEGC across our ensemble, which indicates that we are pointing to the correct finidats. But the right two panels show that TOTVEGC jumps up in the first year of pSASU and then down in the second year, across all members.
![TVC_jumps](https://github.com/user-attachments/assets/214bc349-fa48-4a6f-b58c-84e275f66639)

Finally, it seems like this is not globally uniform, with the jump concentrated within a subset of gridcells:
<img src="https://github.com/user-attachments/assets/fadc1074-0d77-4dba-abfb-1d762e139826" alt="spatial" width="600"/>

Here's the notebook to create those figures:
https://github.com/NCAR/ctsm6_ppe/blob/main/spinup/minioaat_spinup.ipynb

### Important details of your setup / configuration so we can reproduce the bug

Happy to add more details as needed, but mainly this is BGC-nocrop using the 16PFT datasets and our sparsegrid configuration.

Case directories are here for one example ensemble member:
/glade/work/linnia/ctsm5.3.0/cime/scripts/transient/ctsm5.3.0_transient_AD/ctsm5.3.0_transient_AD_oaat0057
/glade/work/linnia/ctsm5.3.0/cime/scripts/transient/ctsm5.3.0_transient_SASU/ctsm5.3.0_transient_SASU_oaat0057
/glade/work/linnia/ctsm5.3.0/cime/scripts/transient/ctsm5.3.0_transient_postSASU/ctsm5.3.0_transient_postSASU_oaat0057
/glade/work/linnia/ctsm5.3.0/cime/scripts/transient/ctsm5.3.0_transient_postSASU/ctsm5.3.0_transient_oaat0057

**Comment by slevis-lmwg** [comment]
@ekluzek and @slevis-lmwg discussed this. I'm pinging @wwieder @olyson since they were in the original emails.

The discontinuities may occur due to the choice of RUN_TYPE. Suggestions to consider:
- AD no change, though is there a discontinuity here, too, according to the GPP timeseries?
- SASU: Erik pointed out that in CTSM testing (SSPMATRIXCN_Ly5_Mmpi-serial.1x1_numaIA.I2000Clm50BgcCropQianRs.izumi_intel.clm-ciso_monthly) we do hybrid for SASU, and I think you do startup. These are likely identical in this case, but a branch may eliminate the discontinuity.
- pSASU: the same CTSM test uses branch for pSASU, so try that.

A complication with branches is that you MUST use the same case. If you want to try this and would like help, let us know. In any case, it seems like a good first thing to try.

**Comment by linniahawkins** [comment]
Thanks for looking into this! I am testing the branch suggestion now and will report back.

**Comment by djk2120** [comment]
So I've looked into this further, and it seems like all of the "jump" is coming from patches where the veg is dead at the end of SASU and then is revitalized to start postSASU. Here's an example:
![patch500_ts](https://github.com/user-attachments/assets/049dbe24-ca6e-4106-a107-8cc43f06ad71)

Is this typical for these types of spinup?

**Comment by dlawrenncar** [comment]
Is reseed dead plants set to true in the postSASU run?  If it is, that
might explain it and probably it should be set to false.

**Comment by linniahawkins** [comment]
Yes, reseed_dead_plants=.true.  - I'll test turning this off next

**Comment by wwieder** [comment]
Ah good. I'd expect this to be an expected feature with `reseed_dead_plants=.true.`.  If you confirm that's the source @linniahawkins maybe we mark this issue as closed with a won't fix?

**Comment by dlawrenncar** [comment]
Depending on how fully systematized the set of spinup steps is, this might
require a change to the default settings for post-SASU.

**Comment by linniahawkins** [comment]
Confirming reseed_dead_plants was the culprit. We're now setting this to false in SASU, postSASU in the PPE spinup workflow. 


![Screen Shot 2024-11-08 at 4 15 19 PM](https://github.com/user-attachments/assets/13516044-72bf-4de8-9baa-78088c9b3d45)

Thanks everyone!

**Comment by olyson** [comment]
I'm guessing reseed_dead_plants=.true. is being set manually in the user_nl_clm for SASU and pSASU in the PPE cases, as this is how it is being set in the recent spinups @slevis-lmwg and I have done, and the PPE is probably following that.   Maybe it should not be set in either SASU or pSASU?  I looked at my earlier PPE cases that were run with the PPE branch and I didn't set it in any of those SASU (then "step3") and pSASU (then "step4") cases.

**Comment by olyson** [comment]
Ok, looks like I posted at the same time @linniahawkins did.

**Comment by wwieder** [comment]
FWIW, I propose:
- We should be reseeding when we start a SASU run (or an AD that's not from a cold-start).  
- I'm less sure reseeding will do anything in postSASU and 
- Reseeding should NOT be done when starting a HIST case.

Not sure if this is something that should be in namelist defaults (or documented in our work flows and tutorials)

**Comment by linniahawkins** [comment]
Agree - we are keeping reseed_dead_plants=.true. in AD (not cold-start). 

And yes Keith, I was setting it to true in user_nl_clm for SASU and postSASU.

**Comment by slevis-lmwg** [comment]
In namelist_defaults_ctsm.xml we currently have
```
<!-- Reseeding of dead plants normally only done for AD spinup -->
<reseed_dead_plants clm_accelerated_spinup="on" use_cn=".true.">.true.</reseed_dead_plants>
<reseed_dead_plants                                            >.false.</reseed_dead_plants>
```

**Comment by slevis-lmwg** [comment]
This confirms @linniahawkins and @olyson's comments that they/we have been setting reseed to true manually for other cases. If we decide to make it automatic for SASU according to @wwieder's comment, then we would add a line in namelist_defaults_ctsm.xml.

**Comment by ekluzek** [comment]
@wwieder @olyson @linniahawkins @slevis-lmwg @djk2120 

Was this resolved or is this something that still needs changes to come in?

**Comment by linniahawkins** [comment]
This has been resolved. Thanks @ekluzek
